src/redisraft.c

```diff
@@ -646,9 +646,32 @@ static void handleRedisCommandAppend(RedisRaftCtx *rr,
         return;
     }
 
-    if (cmd_flags & CMD_SPEC_DENYOOM && (RedisModule_GetUsedMemoryRatio() > 1.0)) {
-        RedisModule_ReplyWithError(ctx, "OOM command not allowed when used memory > 'maxmemory'.");
-        return;
+    /* dry run */
+    for (int i = 0; i < cmds->len; i++) {
+        RaftRedisCommand *cmd = cmds->commands[i];
+        size_t cmdlen;
+        const char *cmdstr = RedisModule_StringPtrLen(cmd->argv[0], &cmdlen);
+        /* skip multi */
+        if (i == 0 && cmdlen == 5 && !strncasecmp(cmdstr, "MULTI", 5)) {
+            continue;
+        }
+
+        char *rm_call_flags;
+        if (RedisModule_GetUsedMemoryRatio() > 1.0) {
```

Comment:
As we are going to call dry-run unconditionally, we don't need to do this check here. rm_call() does more or less same thing, checks usage ratio + one extra if check for DENY_OOM flag. So, I think we can pass "DCEMv" directly. 

Hopefully, we can improve rm_call() performance later.